packages/deploy/README.md

```diff
@@ -39,26 +39,59 @@ where:
-  with or without tags to test just the deployment scripts.
-
-To simplify the execution of the integration tests over forks the following targets are added to the package.json
-file: `fork:mainnet`, `fork:polygon`, `fork:goerli` and `fork:mumbai`.
+coverage. This repo contains integrations tests, tests for the deployment process and tests that verify the integrity of
+the system. For example in the case of the `SignedMultiGiveaway` contract we check the roles and the users assigned to
+them are correctly configured.
+
+To test the deployment process:
+
+- run all the deployment scripts on the hardhat local node just to test them: `yarn deploy`
+- run the deployments scripts over a forks of a live network: `fork:deploy NETWORK` where NETWORK is `mainnet`,`polygon`
+  ,`goerli`,`mumbai`, etc.
+
+The tests the integrity of the system:
+
+- using hardhat local node and `hardhat-deploy` deployment scripts: `yarn test`
+- on a live network over contracts that are already deployed: `yarn test --network NETWORK`. Where NETWORK is `mainnet`
+  , `polygon`, `mumbai`, etc.
+- on a fork of a live network and `hardhat-deploy` deployment scripts: `yarn fork:test NETWORK` where NETWORK
+  is `mainnet`,`polygon`,`goerli`,`mumbai`, etc.
+
+### Tweaking hardhat when forking a live network
+
+We are using some tricks to control the way we run `hardhat` and `hardhat-deploy` when forking live networks. We change
+the `hardhat` configuration depending on some environment variables to be able to run the fork, avoid running some
+deployment scripts or skip the deployment scripts at all. The following table describes the environment variables and
+their use:
+
+| Environment variable                   | Origin            | Description                                         |
+|----------------------------------------|-------------------|-----------------------------------------------------|
+| HARDHAT_FORK=mumbai                    | hardhat.config.ts | configure the hardhat network with forking enabled  |
+| HARDHAT_DEPLOY_ACCOUNTS_NETWORK=mumbai | hardhat-deploy    | use this network for [named accounts](https://github.com/wighawag/hardhat-deploy#1-namedaccounts-ability-to-name-addresses)|
+| HARDHAT_DEPLOY_FIXTURE=true            | hardhat-deploy    | run the deployment scripts before running the tests |
+| HARDHAT_DEPLOY_NO_IMPERSONATION=true   | hardhat-deploy    | Optional. Don't [impersonate unknown accounts](https://hardhat.org/hardhat-network/docs/reference#hardhat_impersonateaccount)|
+| HARDHAT_FORK_INCLUDE_MOCKS=false       | hardhat.config.ts | Optional. Include mock deploy scripts in the fork   |
+| HARDHAT_SKIP_FIXTURES=false            | hardhat.config.ts | Optional. Skip all the deployment scripts           |
+| HARDHAT_FORK_NUMBER=9999999999         | hardhat.config.ts | Optional. Forking block number                      |
+
+There is a `npmScriptHlper` script used in the `package.json` file to simplify the execution of hardhat while setting
```

Comment:
npmScriptHlper > npmScriptHelper

Reply:
I don't think this name reflects very well what this script is doing, maybe:
forkEnv.js
prepareEnv.js
prepareHardhatEnv.js
...

Reply:
One last comment about this script, I was thinking maybe this script could be replaced by a hardhat task ?

Reply:
I think the name is ok: npmScriptHelper it is just to be able to shorten the lines in the npm scripts. 
It cannot be done inside a hardhat task because the env vars affects the hardhat.config file before it is even loaded.